pkg/api/v1alpha1/awsiamconfig_webhook.go

```diff
@@ -74,11 +74,12 @@ func (r *AWSIamConfig) ValidateDelete() error {
 
 func validateImmutableAWSIamFields(new, old *AWSIamConfig) field.ErrorList {
 	var allErrs field.ErrorList
+	specPath := field.NewPath("spec")
 
 	if !new.Spec.Equal(&old.Spec) {
 		allErrs = append(
 			allErrs,
-			field.Invalid(field.NewPath("spec", "AWSIamConfig"), new, "config is immutable"),
+			field.Forbidden(specPath.Child("AWSIamConfig"), "config is immutable"),
```

Comment:
Does `spec.AWSIamConfig` make sense though? Can we somehow just say "AWSIamConfig <name>: config is immutable"?

Reply:
So capi adds the spec path to all these fields we call out, so to stick with capi, we add spec as well. Which means the entire AWS IAM config spec is not mutable, with all the sub fields.